### ISSUE TITLE
Homepage reload glitches

### ISSUE THREAD

**Issue by defagos** [SRGSSR/playsrg-ios#137]
Since a few 3.0.0 betas, we experienced strange homepage animation issues in some cases.

### Issue type

Incorrect behavior

### Environment information

* Application version: 3.0.0
* iOS version: Any
* Device: Any

### Reproducibility

Always reproducible

### Steps to reproduce

Easy with an iPad:

1. Open the videos tab.
1. Start at the top, scroll to the very bottom.
1. Display the last topic detail page.
1. Navigate back by grabbing the left edge of the view, slowly. You can see that the table animates in a very strange way.

**Comment by defagos** [comment]
I could trace the issue to commit a205762609a2d7fd3d842277a56bdfd590d20691 with manual git bissect.

This is very strange since only related to the header height. I cannot debug this for the moment, but slightly tweaking the values fixes the issue. Probably some internal table view calculation which, yielding different results, leads to additional layout changes.

**Comment by defagos** [comment]
I finally found the issue. We recently removed all dummy storyboards (i.e. empty with just a table or collection view), replacing them with manual instantiation. The problem is that table views instantiated in code differ from those instantiated in xibs or storyboards. This is a well-known fact, but what I didn't now is that they differ regarding a very important setting, namely cell sizing.

Their estimated row, header and footer heights are all [dynamic by default when instantiated in code](https://forums.developer.apple.com/thread/81895), not when using a xib or storyboard. We do not use automatic cell sizing (and I personally always had trouble using this feature), and this was leading to content size calculations being made on the fly, making them unreliable in our case (since we calculate row heights ourselves). Printing the content size regularly during scrolling namely reveals that it changes constantly.

This was leading to two issues:

- Position glitches as described above.
- Incorrect behavior when scrolling to the top (by double-tapping a tab). After having switched tabs, the top position could only be reached by tapping many more times on the tab, each double-tap only moving it a bit further.

We recently tweaked the layout constants to mitigate this issue (5cde1d21dc475e9a2c32f7d80d30646ce7e5b9fa), but we knew the issue remained (it seemed less visible, though). I reverted this commit locally to check that with the old values and properly disabled automatic sizing all issue disappear. This is the case, confirming that the automatic cell sizing was the culprit. 

We will keep the new layout constants, though, as they are slightly better anyway. I have introduced a custom `TableView` class which provides proper settings for instantiation in code.

Other issues we recently found in tables (e.g. #137) are not related to this issue. Their fix is still needed.
